src/types.h

```diff
@@ -550,26 +550,54 @@ struct ProbeItem {
 };
 
 const std::vector<ProbeItem> PROBE_LIST = {
-  { "kprobe", { "k" }, ProbeType::kprobe, .show_in_kernel_list = true },
-  { "kretprobe", { "kr" }, ProbeType::kretprobe },
-  { "uprobe", { "u" }, ProbeType::uprobe, .show_in_userspace_list = true },
-  { "uretprobe", { "ur" }, ProbeType::uretprobe },
-  { "usdt", { "U" }, ProbeType::usdt, .show_in_userspace_list = true },
-  { "BEGIN", { "BEGIN" }, ProbeType::special },
-  { "END", { "END" }, ProbeType::special },
-  { "tracepoint", { "t" }, ProbeType::tracepoint, .show_in_kernel_list = true },
-  { "profile", { "p" }, ProbeType::profile },
-  { "interval", { "i" }, ProbeType::interval },
-  { "software", { "s" }, ProbeType::software, .show_in_kernel_list = true },
-  { "hardware", { "h" }, ProbeType::hardware, .show_in_kernel_list = true },
-  { "watchpoint", { "w" }, ProbeType::watchpoint },
-  { "asyncwatchpoint", { "aw" }, ProbeType::asyncwatchpoint },
-  { "kfunc", { "f", "fentry" }, ProbeType::kfunc, .show_in_kernel_list = true },
-  { "kretfunc", { "fr", "fexit" }, ProbeType::kretfunc },
-  { "iter", { "it" }, ProbeType::iter, .show_in_kernel_list = true },
-  { "rawtracepoint",
-    { "rt" },
-    ProbeType::rawtracepoint,
+  { .name = "kprobe",
+    .aliases = { "k" },
+    .type = ProbeType::kprobe,
+    .show_in_kernel_list = true },
```

Comment:
I'm guessing the problem here was that we either have to use named assignment for every field or for no fields?

I'm not sure why we have `show_in_kernel_list` and `show_in_userspace_list` as separate booleans really - it might be nicer to create an enum type that defines what lists to show a probe in, e.g.:
```
enum class ShowInList {
  None,
  Kernel,
  Userspace,
};
```

That way every probe entry can have an explicit listing definition without us having to name every field, e.g.:
```
{ "kprobe", { "k" }, ProbeType::kprobe, ShowInList::Kernel },
```

This might end up being a bigger change than I'm anticipating, so I won't block on this suggestion.

Reply:
Yes, the problem is that you need all-or-nothing in initializer lists, and it makes sense - you can use order or .tags but not really both.